### DISCUSSION TITLE
tasmota compiled gitpod, doesn't keep permanent setting data

### DISCUSSION THREAD

**Discussion by 3N37** [arendst/Tasmota#10811] [Q&A] [answered]
good morning everyone,
 I have compiled with gitpod tasmota with the thermostat option.
 I uploaded the .bin file on a wemos d1mini it works but it doesn't keep the setting at shutdown, I also set the rules (rule 1....) as the manual says but at the shutdown of the wemos all the configuration is lost.
I ask you if there is some trick to get the permanent setting.

**Comment by barbudor** [comment, marked as answer]
Does it keeps your wifi settings or only rules ?
Does it looses the changes only after a power down ? Or also after a `restart 1` ?
Have you typed command `savedata 0` which would have disabled saving to flash ?
=> Type `savedata` without any argument to know what is the currnet value. If 0, restore with `savedata 1`

**Reply by 3N37** [reply]
@barbudor ,
- It keeps the wifi settings because I set them during compilation.

- It does not retain settings entered from the console or GUI.

- It loses configuration data both on reboot and on physical shutdown of wemos.

- I typed  savedata from console and it wrote :

stat/tasmota_9E993AA/RESULT = {"SaveData": "ON"}
so it should be enabled, but it keeps losing settings.

- [ ] the manual writes that but it does not work :
Thermostat persistent storage for configuration~

The thermostat driver stores all configured parameters over MQTT exclusively in RAM, it does not use flash due to the amount of the parameters. This means that at every restart the default parameters will be set again. To avoid this behavior rules can be set-up to reconfigure desired parameters at every restart. See below an example:

ON Power1#boot DO Backlog sensorinputset 1;controllermodeset 2;thermostatmodeset 1;temptargetset %mem1% ENDON
ON mqtt#connected DO Publish2 stat/TestTopic/targetTempValue {"Temp":%mem1%} ENDON
ON mem1#state DO Backlog temptargetset %value%;Publish2 stat/TestTopic/targetTempValue {"Temp":%mem1%} ENDON

**Reply by barbudor** [reply]
Did you compile `tasmota` or `tasmota-minimal` ?
Never use tasmota-minimal as it does not save any parameters. It is intended only to be used as a stepping stone for firmware upgrade of large binaries

Custom compile should always be based on `tasmota` if you intend to modify the feature list.
If you just set wifi/IP/Mqtt settings and some setoptions and are looking to reduce binary size, you can use `tasmota-lite`.

If the above is not the reasons for the problem, I suggest you join on Discord. It will be easier than here to make interactive debugging suggestion.

**Reply by 3N37** [reply]
ok thanks barbudor,
The version of tasmota 9.2 ,defines and wifi are available in tasmocompiler and can be set with a simple form for the rest is all automatic.

**Reply by benzino77** [reply]
https://github.com/benzino77/tasmocompiler/discussions/123

**Reply by 3N37** [reply]
I solved the problem by compiling tasmota for a system with 1MB of ram and not 4MB as wemos has.
My aim is to compile tasmota with thermostat options so that it is independent and works in case of a home automation block.
I still have this problem: the rules

 ON Power1#boot DO Backlog sensorinputset 1;controllermodeset 2;thermostatmodeset 1;temptargetset %mem1% ENDON

but even setting from the console e.g.: controllermodeset 2
are not saved;

while commands given from the GUI and some console commands such as:mem1 28, are saved correctly.

Does anyone know why?
